src/main/java/se/sundsvall/messaging/integration/party/PartyClient.java

```diff
@@ -0,0 +1,22 @@
+package se.sundsvall.messaging.integration.party;
+
+import org.springframework.cloud.openfeign.FeignClient;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+
+@FeignClient(
+	name = PartyIntegration.INTEGRATION_NAME,
+	url = "${integration.party.base-url}",
+	configuration = PartyIntegrationConfiguration.class,
+	dismiss404 = true
+)
+public interface PartyClient {
+
+	@GetMapping("/{municipalityId}/{type}/{partyId}/legalId")
+	ResponseEntity<String> getLegalIdByPartyId(
```

Comment:
Change to Optional<String> here will let you write:
```
		return partyClient.getLegalId(PRIVATE, partyId)
			.or(() -> partyClient.getLegalId(ENTERPRISE, partyId))
			.orElse(null);
```
in the PartyIntegration-class. If a 404 is thrown an Optional.empty() will be returned instead.